### PR TITLE
#20777 Sync "Show virtual model editor" with active navigator view mode

### LINKED ISSUE #20777: Impossible to toggle navigator view Custom-Advanced
### Description

https://github.com/dbeaver/dbeaver/assets/84867187/8b5bea66-1207-4597-9eec-ef35cbbbc1c8

### DBeaver Version

23.1.3

### Operating System

win 10

### Database and driver

_No response_

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java [hunk 1]
```diff
@@ -634,8 +649,6 @@ public void saveSettings(DBPDataSourceContainer dataSource) {
         }
         dsDescriptor.setNavigatorSettings(this.navigatorSettings);
 
-        dsDescriptor.getNavigatorSettings().setHideVirtualModel(!this.showVirtualModelCheck.getSelection());
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java:637]: Why remove?

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java [hunk 2]
```diff
@@ -282,6 +287,10 @@ public void createControl(Composite parent)
     {
         boldFont = UIUtils.makeBoldFont(parent.getFont());
 
+        if (navigatorSettings == null) {
+            navigatorSettings = new DataSourceNavigatorSettings(getWizard().getSelectedNavigatorSettings());
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java:291]: What the point having non-associated nav settings? They must be in ds descriptor.

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java:637]: It's updated when the checkbox changes its state.

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java (hunk 2 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java:291]: They're assigned in `saveSettings` method.

**Comment by MashaKorax** [comment]
verified

**Pull request merged**